src/user-event/scroll/__tests__/scrollTo.test.tsx

```diff
@@ -0,0 +1,265 @@
+    const { scrollView, events } = renderScrollViewWithToolkit();
+    const user = userEvent.setup();
+
+    await user.scrollTo(scrollView, { y: [0, 33, 67, 100] });
+    expect(pickEventEssentials(events)).toEqual([
+      ['scrollBeginDrag', 0, 0],
+      ['scroll', 33, 0],
+      ['scroll', 67, 0],
+      ['scrollEndDrag', 100, 0],
+    ]);
+    expect(events).toMatchSnapshot('scrollTo({ y: [0, 33, 67, 100] })');
+  });
+
+  it('supports vertical momentum scroll', async () => {
+    const { scrollView, events } = renderScrollViewWithToolkit();
+    const user = userEvent.setup();
+
+    await user.scrollTo(scrollView, { momentumY: [0, 10, 20] });
+    expect(pickEventEssentials(events)).toEqual([
+      ['momentumScrollBegin', 0, 0],
+      ['scroll', 10, 0],
+      ['scroll', 20, 0],
+      ['momentumScrollEnd', 20, 0],
+    ]);
+    expect(events).toMatchSnapshot('scrollTo({ momentumY: [0, 10, 20] })');
+  });
+
+  it('supports vertical drag & momentum scroll', async () => {
+    const { scrollView, events } = renderScrollViewWithToolkit();
+    const user = userEvent.setup();
+
+    await user.scrollTo(scrollView, { y: 10, momentumY: 20 });
+    expect(pickEventEssentials(events)).toEqual([
+      ['scrollBeginDrag', 0, 0],
+      ['scroll', 2, 0],
+      ['scroll', 4, 0],
+      ['scroll', 6, 0],
+      ['scroll', 8, 0],
+      ['scrollEndDrag', 10, 0],
+      ['momentumScrollBegin', 10, 0],
+      ['scroll', 12, 0],
+      ['scroll', 14, 0],
+      ['scroll', 16, 0],
+      ['scroll', 18, 0],
+      ['scroll', 20, 0],
+      ['momentumScrollEnd', 20, 0],
+    ]);
+  });
+
+  it('supports horizontal drag scroll', async () => {
+    const { scrollView, events } = renderScrollViewWithToolkit({
+      horizontal: true,
+    });
+    const user = userEvent.setup();
+
+    await user.scrollTo(scrollView, { x: 100 });
+    expect(pickEventEssentials(events)).toEqual([
+      ['scrollBeginDrag', 0, 0],
+      ['scroll', 0, 20],
+      ['scroll', 0, 40],
+      ['scroll', 0, 60],
+      ['scroll', 0, 80],
+      ['scrollEndDrag', 0, 100],
+    ]);
+    expect(events).toMatchSnapshot('scrollTo({ x: 100 })');
+  });
+
+  it('supports horizontal drag scroll with explicit steps', async () => {
+    const { scrollView, events } = renderScrollViewWithToolkit({
+      horizontal: true,
+    });
+    const user = userEvent.setup();
+
+    await user.scrollTo(scrollView, { x: [15, 30, 45, 60] });
+    expect(pickEventEssentials(events)).toEqual([
+      ['scrollBeginDrag', 0, 15],
+      ['scroll', 0, 30],
+      ['scroll', 0, 45],
+      ['scrollEndDrag', 0, 60],
+    ]);
+    expect(events).toMatchSnapshot('scrollTo({ x: [15, 30, 45, 60] })');
+  });
+
+  it('supports horizontal momentum scroll', async () => {
+    const { scrollView, events } = renderScrollViewWithToolkit({
+      horizontal: true,
+    });
+    const user = userEvent.setup();
+
+    await user.scrollTo(scrollView, { momentumX: [0, 10, 20] });
```

Comment:
you can't have a scroll with only momentum no? I think this should be allowed only when x is also defined

Reply:
Let's discuss that within the [API proposal](https://github.com/callstack/react-native-testing-library/issues/1450#issuecomment-1687756141)

Reply:
Changed. Now the drag part (`y` or `x`) is mandatory, while momentum part is optional.